tests/kohana/RequestTest.php

```diff
@@ -736,10 +736,60 @@ public function test_passes_client_params()
 		$this->assertEquals($client->strict_redirect(), FALSE);
 	}
 
+	/**
+	 * Tests correctness request content-length header after calling render
+	 */
+	public function test_content_length_after_render()
+	{
+		$request = Request::factory('https://example.org/post')
+			->client(new Request_Client_Content_Length_Kohana_RequestTest)
+			->method(Request::POST)
+			->post(array('aaa' => 'bbb'));
+
+		$request->render();
+
+		$response = $request->execute();
+
+		$this->assertEquals(strlen($request->body()), $response->body());
```

Comment:
@enov I found this assertion a bit hard to understand at first, without going to look at the implementation of `Request_Client_Content_Length_Kohana_RequestTest`. It might become hard to maintain once the context of the commit/PR is not available.

It might be better to use a `Kohana_RequestTest_Headers_Echoing_Request_Client` and an assertion like `$this->assertEquals(strlen($request->body()), json_decode($response->body(), TRUE)['content-length'])`.

Or better yet a `Kohana_RequestTest_Header_Spying_Request_Client_External` and an assertion like `$this->assertEquals(strlen($request->body()), $client->getReceivedRequestHeaders('content-length'))` ?

Would also make that mock more reusable for other tests in future.